### PR TITLE
Support different service Profiles in HAproxy

### PR DESCRIPTION
**Pull request by relistan**
This change allows Sidecar to better support different kinds of service configurations in HAproxy. The approach taken keeps the data exchanged over gossip to a minimum by providing a single service profile setting that describes a whole profile of settings to be used in HAproxy instead of sending around a lot of gossip packets containing settings. 

This change:

* Removes the `ProxyMode` setting in favor of `Profile`
* Removes the `ProxyMode` Docker label in favor of `ServiceProfile`
* Uses the `Profile` settings to identify a pair of frontend and backend HAproxy config templates to use for this service
* Provides tests and examples to show how this is used to modify HAproxy behavior for a service

**NOTE** This is a **BREAKING** change because it removes the `ProxyMode` setting. This could have been preserved, but it's entirely supplanted by the new settings so I have removed it. We are not using it and I don't currently know of anyone who is using it. As a result, however, I will do a release from this code and bump the version to 1.1.

cc @mosche @benoahriz @mjvaldez @actaeon

### CODE REVIEW

**Comment by mosche** [comment]
👀 , but have to get back to it tomorrow

**Comment by benoahriz** [comment]
this looks great.  I didnt get it up and running locally but I'm planning on testing it out.  Tests look good only thing I can think of would be to add some documentation on how and why you might use this.  I'd be happy to do that if you think its a good idea @relistan. 👍

**Comment by relistan** [comment]
@benoahriz good point. Added some docs and cleaned up last remaining refs to `ProxyMode`.

**Comment by relistan** [comment]
Once this is merged, I'll pull the changes into [HAproxy-API](https://github.com/nitro/haproxy-api) as well.

**Comment by relistan** [comment]
I plan to merge this today or tomorrow. Speak now or forever wish you had. ;)

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -2,179 +2,22 @@
 	"ImportPath": "github.com/newrelic/sidecar",
```
Comment (mihaitodor) [inline Godeps/Godeps.json]: Looks like Godeps got messed up...

**Comment by relistan** [comment]
We've ended up not needing this and it complicates things enough I'm not going to merge it. Closing.

**Pull request closed without merging**